src/ray/rpc/client_call.h

```diff
@@ -67,6 +83,16 @@ class ClientCallImpl : public ClientCall {
   /// gRPC status of this request.
   grpc::Status status_;
 
+  /// Mutex to protect the return_status_ field.
+  absl::Mutex mutex_;
+
+  /// This is the status to be returned from GetStatus(). It is safe
+  /// to read from other threads while they hold mutex_. We have
+  /// return_status_ = GrpcStatusToRayStatus(status_) but need
+  /// a separate variable because status_ is set internally by
+  /// GRPC and we cannot control it holding the lock.
+  ray::Status return_status_ GUARDED_BY(mutex_);
```

Comment:
Would these changes work on GCC? Or are we implicitly only supporting Clang moving forward?

There are pros and cons to both sides (I don't know what the right answer is), but if the intention is to drop support for non-Clang compilers, it might be worth discussing this explicitly with others (if it hasn't been discussed already) and then maybe mentioning it somewhere in the docs rather than just having it fail implicitly.

Reply:
This works (i.e. compiles) in GCC. Not sure if it does the safety analysis thought.